### PR TITLE
Fix DNS record parsing and add SRV record support.

### PR DESCRIPTION
**Pull request by NN---**
https://en.wikipedia.org/wiki/SRV_record

### CODE REVIEW

File: network/dns_packet.ksy [hunk 1]
```diff
@@ -60,19 +60,24 @@ types:
...
+            'type_type::txt': txt_body
+
   domain_name:
     seq:
       - id: name
         type: label
         repeat: until
         doc: "Repeat until the length is 0 or it is a pointer (bit-hack to get around lack of OR operator)"
-        repeat-until: "_.length == 0 or _.length == 0b1100_0000"
+        repeat-until: "_.length == 0 or _.length >= 192"
```
Comment (GreyCat) [inline network/dns_packet.ksy]: Shouldn't it be `_.length == 0 or _.is_pointer` then, as we already have this `is_pointer` set up?

File: network/dns_packet.ksy [hunk 2]
```diff
@@ -89,7 +94,7 @@ types:
         size: length
     instances:
       is_pointer:
-        value: length == 0b1100_0000
+        value: length >= 192
```
Comment (GreyCat) [inline network/dns_packet.ksy]: Can you explain why this change was necessary?

File: network/dns_packet.ksy (hunk 1 above)
Comment (GreyCat) [inline network/dns_packet.ksy]: Shouldn't it be `_.length == 0 or _.is_pointer` then, as we already have this `is_pointer` set up?

File: network/dns_packet.ksy (hunk 2 above)
Comment (NN---) [reply network/dns_packet.ksy]: https://tools.ietf.org/html/rfc1035
Offset can be non zero:

```
    +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
    | 1  1|                OFFSET                   |
    +--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+--+
```

File: network/dns_packet.ksy (hunk 1 above)
Comment (NN---) [reply network/dns_packet.ksy]: Fixed.

**Comment by NN---** [comment]
Ping. Can this be merged ?

File: network/mdns_packet.ksy
Comment (generalmimon) [inline network/mdns_packet.ksy:1]: Why did you add this duplicate of `dns_packet.ksy`?

**Comment by generalmimon** [comment]
> Ping. Can this be merged ?

Not with merge conflicts.

File: network/mdns_packet.ksy
Comment (generalmimon) [reply network/mdns_packet.ksy:1]: I see that you only changed the `encoding: ASCII` to `encoding: UTF-8`, but that is certainly not a sufficient reason to create a duplicate file. Given that UTF-8 is a superset of ASCII, why not just switch to `encoding: UTF-8` in the main `dns_packet.ksy`?

Comment (NN---) [reply network/mdns_packet.ksy:1]: If this acceptable then it can be one file.

**Comment by generalmimon** [comment]
@NN--- Are you no longer interested in getting this merged?

**Comment by NN---** [comment]
> @NN--- Are you no longer interested in getting this merged?

I see there a re already several changes in the file.
Not sure whether my change is needed anymore.
